src/include/hardware.h

```diff
@@ -73,6 +73,38 @@
 #define WT_WRITE_ONCE(v, val) WT_RELEASE_WRITE_WITH_BARRIER(v, val)
 #endif
 
+/*
+ * The below assembly implements the read-acquire semantic. The if branches get removed at compile
+ * time as the sizeof instruction evaluates at compile time.
+ *
+ * Depending on the size of the given type we choose the appropriate ldapr variant, additionally the
+ * W register variants are used if possible which map to the lower word of the associated X
+ * register.
+ *
+ * Finally the "Q" constraint is used for the given input operand, this instructs the compiler to
+ * generate offset free ldapr instructions. ldapr instructions, prior to version RCpc 3, don't
+ * support offsets.
+ */
+#ifdef HAVE_RCPC
+#define WT_ACQUIRE_READ(v, val)                                      \
+    do {                                                             \
+        if (0) {                                                     \
```

Comment:
Please add a comment explaining how this gives us type checking even when the code is unreachable. This'll confuse future readers.

Reply:
258c94684a42c45e9d796386376386c39fce564e